src/SamplesApp/SamplesApp.UITests/MessageDialogTests/MessageDialogTest.cs

```diff
@@ -0,0 +1,30 @@
+﻿using NUnit.Framework;
+using SamplesApp.UITests.TestFramework;
+using Uno.UITest.Helpers;
+using Uno.UITest.Helpers.Queries;
+
+namespace SamplesApp.UITests.MessageDialogTests
+{
+	[TestFixture]
+	public partial class MessageDialogTest : SampleControlUITestBase
+	{
+		[Test]
+		[AutoRetry]
+		public void When_Click_Outside_Dialog_Expect_No_Dismiss()
+		{
+			Run("UITests.MessageDialog.MessageDialogTest");
+			var button = _app.Marked("showDialogBtn");
+			_app.WaitForElement(button);
+			button.Tap();
+
+			using var screenshot = TakeScreenshot("BeforeClicking");
+
+			// TODO: Click outside.
+
+			using var screenshot2 = TakeScreenshot("AfterClicking");
+
+			// TODO: Verify screenshots are the same.
```

Comment:
You cane use `ImageAssert` for this, eg:
https://github.com/unoplatform/uno/blob/3bff3470b445c91dd2a1150a7b90d218cc7a878b/src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/BorderTests/Border_Tests.cs#L30-L32

Reply:
@davidjohnoliver Thanks! I have been looking at extensions methods on `ScreenshotInfo`.